spec/event_spec.rb

```diff
@@ -0,0 +1,141 @@
+    let(:event) { Event.new("Carla's Craft Connection", [sewing, knitting], [hector, toni])}
+    
+    describe '#attendee_names' do
+      it 'can return array of person names' do
+        expect(event.attendee_names).to eq(['Hector', 'Toni'])
+      end
+    end
+    
+    describe '#craft_with_most_supplies' do
+      it 'can return the craft with the most supplies' do
+        expect(event.craft_with_most_supplies).to eq('sewing')
+      end
+    end
+    
+    describe '#supply_list' do
+      it 'can return an array of all required supplies at event' do
+        expected = ['fabric', 'scissors', 'thread', 'sewing_needles', 'yarn', 'knitting_needles']
+        expect(event.supply_list).to eq(expected)
+      end
+    end
+  end
+  
+  describe 'iteration3' do
+    let(:hector) { Person.new({name: 'Hector', interests: ['sewing', 'millinery', 'drawing']}) }
+    let(:toni) { Person.new({name: 'Toni', interests: ['sewing', 'knitting']}) }
+    let(:tony) { Person.new({name: 'Tony', interests: ['drawing', 'knitting']}) }
+
+    let(:knitting) { Craft.new('knitting', {yarn: 20, scissors: 1, knitting_needles: 2}) }
+    let(:sewing) { Craft.new('sewing', {fabric: 5, scissors: 1, thread: 1, sewing_needles: 1}) }
+    let(:painting) { Craft.new('painting', {canvas: 1, paint_brush: 2, paints: 5}) }
+    let(:event) { Event.new("Carla's Craft Connection", [knitting, painting, sewing], [hector, toni, tony])}
+    
+    describe '#attendees_by_craft_interest' do
+      it 'can return hash with craft name keys, and interested persons array' do
+        expected = {
+                    "knitting"=> [toni, tony],
+                    "painting"=> [],
+                    "sewing"=> [hector, toni]
+                    }
+        expect(event.attendees_by_craft_interest).to eq(expected) 
+      end
+    end
+
+    describe '#crafts_that_use' do
+      it 'can return array of crafts that use a specific supply' do
+        expected = [knitting, sewing]
+        expect(event.crafts_that_use('scissors')).to eq(expected)
+        expect(event.crafts_that_use('fire')).to eq([])
+      end
+    end
+  end
+
+  # describe 'iteration4' do
+  #   let(:hector) { Person.new({name: 'Hector', interests: ['sewing', 'millinery', 'drawing', 'painting']}) }
+  #   let(:toni) { Person.new({name: 'Toni', interests: ['sewing', 'knitting']}) }
+  #   let(:tony) { Person.new({name: 'Tony', interests: ['drawing', 'knitting', 'painting']}) }
+
+  #   let(:knitting) { Craft.new('knitting', {yarn: 20, scissors: 1, knitting_needles: 2}) }
+  #   let(:sewing) { Craft.new('sewing', {fabric: 5, scissors: 1, thread: 1}) }
+  #   let(:painting) { Craft.new('painting', {canvas: 1, paint_brush: 2, paints: 5}) }
+  #   let(:event) { Event.new("Carla's Craft Connection", [knitting, painting, sewing], [hector, toni, tony])}
+
+  #   # describe '#can_build_crafts_list' do
+
+  #   # end
+  #   describe '#assign_attendees_to_crafts' do
```

Comment:
This is a bit harder to test when we are returning random results. It may make sense to test individual lists than the combination of all lists if that makes sense.